rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -645,6 +659,43 @@ public void onRemoval(RemovalNotification<String, FileMetadata> notification)
     }
   }
 
+  private class ClusterManagerMetadata
```

Comment:
Why this separate class? Isnt ClusterManager enough?

Reply:
We need list of nodes and currentnodeindex. although these could be derived from clustermanager itself, (like `clustermanager.getNodes()` and `clustermanager.getNodes().getIndexOf(currentNodeName)`) thought of storing these info for use in other sections.

Reply:
Seems to be just bloating code as of now. Unless there is a need of it we can get rid of it.